### PR TITLE
[#1703] expose org.eclipse.xtext.preferences as api

### PR DESCRIPTION
**Pull request by cdietrich**
[#1703] expose org.eclipse.xtext.preferences as api
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by szarnekow** [comment]
Just checked the code in that package. Looks good in general. One thing that bugs me is the error handling in the concrete incarnations of `TypedPreferenceKey`, e.g. `IntegerKey` will log a warning in case of `NumberFormatException` and return the default value, `BooleanKey` will use the conversion rules of `Boolean.valueOf` thus often not using the default value even though the value string may be not a boolean representation. StringKey will happily return null instead of the default value if the given string is null. It's not really consistent.

Long story short: Should we try to make this a little better before we publish it as API in its current shape?

**Comment by cdietrich** [comment]
good question. i did not really look into it.

**Review by szarnekow — APPROVED** [review]
